plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -70,11 +70,13 @@ export type CreateWorkerOptions = {
  * @public
  */
 export class TaskWorker {
-  private constructor(private readonly options: TaskWorkerOptions) {}
+  private taskQueue: PQueue;
 
-  private taskQueue: PQueue = new PQueue({
-    concurrency: this.options.concurrentTasksLimit,
-  });
+  private constructor(private readonly options: TaskWorkerOptions) {
+    this.taskQueue = new PQueue({
+      concurrency: options.concurrentTasksLimit,
+    });
```

Comment:
Ran into issues with this when running `backend-next`, either way I feel this is cleaner